src/modules/Builder/features/SaveAndPublish/SaveAndPublish.utils.ts

```diff
@@ -142,18 +179,22 @@ const getScore = (score: ScoreReport, items: ActivityFormValues['items']) => ({
   conditionalLogic: score.conditionalLogic?.map((conditional) => ({
     ...conditional,
     ...removeReportsFields(),
-    conditions: getConditions(items, conditional.conditions),
+    conditions: getConditions(items, conditional.conditions, score),
   })),
 });
 
-const getSection = (section: SectionReport, items: ActivityFormValues['items']) => ({
+const getSection = (
+  section: SectionReport,
+  items: ActivityFormValues['items'],
+  scores?: ScoreReport[],
```

Comment:
could you please use an object for parameters instead of sequence of parameters?